lib/ssl/src/ssl.erl

```diff
@@ -2233,6 +2242,27 @@ opt_handshake(UserOpts, Opts, _Env) ->
 
     Opts#{handshake => HS, max_handshake_size => MHSS}.
 
+opt_use_srtp(UserOpts, #{protocol := Protocol} = Opts, _Env) ->
```

Comment:
Should we not make the user option a map here to reduce the number of options?
{use_strp, #{mki := Mki, protection_profiles := PPs} 

When included in Opts it shall be a map,  and it shall not be included in Opts if it is not enabled.
We are trying to reduce the size of Opts map.

Reply:
Reducing the size of Opts is good.  
Are there any options implemented this way?  
At first I was going to implement options exactly as you propose, but then decided to do it more consistently with majority of other options (or all of them?).

Reply:
With a map as input?
  Only certs_keys comes to mind currently (but we want to change ocsp to be a map).
  
 Search for set_opt_new for example of optionally setting Opts map.

Reply:
Looks good to me, tough a question should we translate use_srtp record in the internal options directly?
Then we don't need the translation in "runtime".